src/forms.rs

```diff
@@ -76,8 +77,211 @@ impl<'a> Field<'a> {
+    /// [`Field::choice_options`] array of the first visible option for
+    /// scrollable lists.
+    pub fn choice_top_index(&mut self, index: i32) -> &mut Self {
+        self.dict.pair(Name(b"TI"), index);
+        self
+    }
+
+    /// Start writing the `/I` array to set the indices of the currently
+    /// selected options. The integers in this array must be sorted in ascending
+    /// order and correspond to 0-based indices in the [`Field::choice_options`]
+    /// array.
+    ///
+    /// This entry shall be used for choice fields which allow multiple
+    /// selections ([`FieldFlags::MULTI_SELECT`]). This means when two or more
+    /// elements in the [`Field::choice_options`] array have different names
+    /// but export the same value or when the value fo the choice field is an
+    /// array. This entry should not be used for choice fields that do not allow
+    /// multiple selections. PDF 1.4+.
+    pub fn choice_indices(&mut self) -> TypedArray<'_, i32> {
+        self.dict.insert(Name(b"I")).array().typed()
+    }
+
+    /// Write the `/V` attribute to set the currently selected values
+    /// of this choice field. Should be one of the values given in
+    /// [`Self::choice_options`] or `None` if no choice is selected. Only
+    /// permissible on choice fields.
+    pub fn choice_value<'b>(&mut self, option: Option<TextStr>) -> &mut Self {
+        match option {
+            Some(value) => self.dict.pair(Name(b"V"), value),
+            None => self.dict.pair(Name(b"V"), Null),
+        };
+        self
+    }
+
+    /// Write the `/V` attribute to set the currently selected values of this
+    /// choice field. See also [`Self::choice_value`], for a single or no value.
+    /// Only permissible on choice fields.
+    pub fn choice_values<'b>(
+        &mut self,
+        options: impl IntoIterator<Item = TextStr<'b>>,
+    ) -> &mut Self {
+        self.dict.insert(Name(b"V")).array().items(options);
+        self
+    }
+
+    /// Write the `/DV` attribute to set the default selected value
+    /// of this choice field. Should be one of the values given in
+    /// [`Self::choice_options`] or `None` if no choice is selected. Only
+    /// permissible on choice fields.
+    pub fn choice_default_value(&mut self, option: Option<TextStr>) -> &mut Self {
+        match option {
+            Some(value) => self.dict.pair(Name(b"DV"), value),
+            None => self.dict.pair(Name(b"DV"), Null),
+        };
+        self
+    }
+
+    /// Write the `/DV` attribute to set the default selected values of this
+    /// choice field. See also [`Self::choice_default_value`], for a single or
+    /// no value. Only permissible on choice fields.
+    pub fn choice_default_values<'b>(
+        &mut self,
+        options: impl IntoIterator<Item = TextStr<'b>>,
+    ) -> &mut Self {
+        self.dict.insert(Name(b"DV")).array().items(options);
+        self
+    }
+}
+
+/// Only permissible on button fields.
```

Comment:
Why are buttons below the other stuff when they come first in the spec?

Reply:
I'll rearrange it to fit the spec and the rest of the code base better.